src/activities/ActivityUsageEntity.js

```diff
@@ -86,41 +86,105 @@ export class ActivityUsageEntity extends Entity {
 		activityCollectionHref && this._subEntity(ActivityUsageCollectionEntity, activityCollectionHref, onChange);
 	}
 
-	_getDueDateSubEntity() {
-		return this._entity
-			&& this._entity.getSubEntityByClass(Classes.dates.dueDate);
-	}
-
 	/**
 	 * @returns {string} Due date of the activity usage
 	 */
 	dueDate() {
-		const dueDate = this._getDueDateSubEntity();
+		const dueDate = this._getDateSubEntity(Classes.dates.dueDate);
 		return dueDate && dueDate.properties.date;
 	}
 
 	/**
 	 * @returns {bool} Whether or not the edit due date action is present on the activity usage entity
 	 */
 	canEditDueDate() {
-		const dueDate = this._getDueDateSubEntity();
-		return (dueDate && dueDate.hasActionByName(Actions.activities.update))
-			|| this._entity.hasActionByName(Actions.activities.startAddNew);
+		return this._canEditDate( Classes.dates.dueDate )
 	}
 
 	/**
  	 * Updates the due date of the activity usage entity to the date specified
-	 * @param {string} dateString Date string to set as the due date, or empty string to clear the due date
+	 * @param {string} dateValue Date string to set as the due date, or empty string to clear the due date
 	 */
-	async setDueDate(dateString) {
-		if (!this.canEditDueDate) {
+	async setDueDate(dateValue) {
+		await this._setDate(dateValue, Classes.dates.startDate, 'dueDate');
+	}
+
+	/**
+	 * @returns {string} Start date of the activity usage
+	 */
+	startDate() {
+		const startDate = this._getDateSubEntity(Classes.dates.startDate);
+		return startDate && startDate.properties.date;
+	}
+
+	/**
+	 * @returns {bool} Whether or not the edit start date action is present on the activity usage entity
+	 */
+	canEditStartDate() {
+		return this._canEditDate( Classes.dates.startDate )
+	}
+
+	/**
+ 	 * Updates the start date of the activity usage entity to the date specified
+	 * @param {string} dateValue Date string to set as the start date, or empty string to clear the start date
+	 */
+	async setStartDate(dateValue) {
+		await this._setDate(dateValue, Classes.dates.startDate, 'startDate');
+	}
+
+	/**
+	 * @returns {string} End date of the activity usage
+	 */
+	endDate() {
+		const endDate = this._getDateSubEntity(Classes.dates.endDate);
+		return endDate && endDate.properties.date;
+	}
+
+	/**
+	 * @returns {bool} Whether or not the edit end date action is present on the activity usage entity
+	 */
+	canEditEndDate() {
+		return this._canEditDate( Classes.dates.endDate )
+	}
+
+	/**
+ 	 * Updates the end date of the activity usage entity to the date specified
+	 * @param {string} dateValue Date string to set as the end date, or empty string to clear the end date
+	 */
+	async setEndDate(dateValue) {
+		await this._setDate( dateValue, Classes.dates.endDate, 'endDate');
+	}
+
+	_getDateSubEntity(dateClass) {
+		return this._entity
+			&& this._entity.getSubEntityByClass(dateClass);
+	}
+
+	/**
+	 * @param {string} dateClass The class for the date entity
```

Comment:
Not a biggie, but JSDoc-ing these private-ish methods will surface them in the docs and potentially make people use them - I'd say either make them intentionally public-ish, or just leave them undocumented. (Admittedly though, this is a feeble attempt to add access modifiers to JS, which... nope.)